master/internal/rm/kubernetesrm/pods.go

```diff
@@ -917,26 +917,25 @@ func (p *pods) cleanUpPodHandler(ctx *actor.Context, podHandler *actor.Ref) erro
 func (p *pods) handleAPIRequest(ctx *actor.Context, apiCtx echo.Context) {
 	switch apiCtx.Request().Method {
 	case echo.GET:
-		summary, err := p.summarize(ctx)
-		if err != nil {
-			ctx.Respond(apiCtx.JSON(http.StatusInternalServerError, err))
+		summaries := p.summarizeClusterByNodes(ctx)
+		_, nodesToPools := p.getNodeResourcePoolMapping(summaries)
+		for nodeName, summary := range summaries {
+			summary.ResourcePool = nodesToPools[summary.ID]
+			summaries[nodeName] = summary
```

Comment:
kinda awkward that summary is not a pointer here

Reply:
Counterpoint: returning a slice of pointers is kinda awkward.

Reply:
why is that? i just thought it was awkward to need to mutate it this way since `summary` is a shallow copy.